src/RestApi/StoreApi/Schemas/AbstractSchema.php

```diff
@@ -64,6 +64,95 @@ function( $property ) {
 		);
 	}
 
+	/**
+	 * Returns consistent currency schema used across endpoints for prices.
+	 *
+	 * @return array
+	 */
+	protected function get_store_currency_properties() {
+		return [
+			'currency_code'               => [
+				'description' => __( 'Currency code (in ISO format) for returned prices.', 'woo-gutenberg-products-block' ),
+				'type'        => 'string',
+				'context'     => [ 'view', 'edit' ],
+				'readonly'    => true,
+			],
+			'currency_symbol'             => [
+				'description' => __( 'Currency symbol for the currency which can be used to format returned prices.', 'woo-gutenberg-products-block' ),
+				'type'        => 'string',
+				'context'     => [ 'view', 'edit' ],
+				'readonly'    => true,
+			],
+			'currency_minor_unit'         => [
+				'description' => __( 'Currency minor unit (number of digits after the decimal separator) for returned prices.', 'woo-gutenberg-products-block' ),
+				'type'        => 'integer',
+				'context'     => [ 'view', 'edit' ],
+				'readonly'    => true,
+			],
+			'currency_decimal_separator'  => array(
+				'description' => __( 'Decimal separator for the currency which can be used to format returned prices.', 'woo-gutenberg-products-block' ),
+				'type'        => 'string',
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+			),
+			'currency_thousand_separator' => array(
+				'description' => __( 'Thousand separator for the currency which can be used to format returned prices.', 'woo-gutenberg-products-block' ),
+				'type'        => 'string',
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+			),
+			'currency_prefix'             => array(
+				'description' => __( 'Price prefix for the currency which can be used to format returned prices.', 'woo-gutenberg-products-block' ),
+				'type'        => 'string',
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+			),
+			'currency_suffix'             => array(
+				'description' => __( 'Price prefix for the currency which can be used to format returned prices.', 'woo-gutenberg-products-block' ),
+				'type'        => 'string',
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+			),
+		];
+	}
+
+	/**
+	 * Prepares a list of store currency data to return in responses.
+	 *
+	 * @return array
+	 */
+	protected function get_store_currency_response() {
+		$position = get_option( 'woocommerce_currency_pos' );
+		$symbol   = html_entity_decode( get_woocommerce_currency_symbol() );
+		$prefix   = '';
+		$suffix   = '';
+
+		switch ( $position ) {
+			case 'left_space':
+				$prefix = $symbol . ' ';
+				break;
+			case 'left':
+				$prefix = $symbol;
+				break;
+			case 'right_space':
+				$suffix = ' ' . $symbol;
+				break;
+			case 'right':
```

Comment:
These cases would be best defined as constants.  I realize this is likely a Woo Core thing though so no need to block this pull.

Reply:
Adding todo